TESTS/mbedmicro-rtos-mbed/malloc/main.cpp

```diff
@@ -14,75 +14,161 @@
  * limitations under the License.
  */
 #include "mbed.h"
-#include "test_env.h"
-#include "rtos.h"
+#include "greentea-client/test_env.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
+
 
 #if defined(MBED_RTOS_SINGLE_THREAD)
   #error [NOT_SUPPORTED] test not supported
 #endif
 
-#define NUM_THREADS         5
+using utest::v1::Case;
+
+extern uint32_t mbed_heap_size;
+static const int test_timeout = 25;
+volatile bool thread_should_continue = true;
+#define NUM_THREADS         4
```

Comment:
Did the number of threads need to be decreased to pass testing?

Reply:
We could restore back thread count to 5, but it will cause fail on GCC_ARM.
To run 5 threads on GCC_ARM we have to free additional ~300B of heap memory.

To achieve this we could reduce thread allocation to 10B or move part of Thread objects on stack (but can not move all of them because we reach main thread stack limit).

Reply:
Why do 5 threads no longer fit?

Reply:
it turns out that refactoring (new test cases, functions, ...) increased static RAM usage by 480B as a result less memory remains for heap

```
python tools/memap.py BUILD/tests/nucleo_f070rb/GCC_ARM/TESTS/mbedmicro-rtos-mbed/malloc/malloc.map -t GCC_ARM

befor refactoring
Total Static RAM memory (data + bss): 10172 bytes

after refactoring
Total Static RAM memory (data + bss): 10652 bytes

diff 
Total Static RAM memory (data + bss): 480 bytes
```

```
extern uint32_t mbed_heap_size;
printf("mbed_heap_size: %lu  \r\n", mbed_heap_size);

befor refactoring
mbed_heap_size: 4996

after refactoring
mbed_heap_size: 4516

diff 
mbed_heap_size: 480
```